sdk/state/update.go

```diff
@@ -0,0 +1,140 @@
+package state
+
+import (
+	"errors"
+	"time"
+
+	"github.com/stellar/experimental-payment-channels/sdk/txbuild"
+	"github.com/stellar/go/xdr"
+)
+
+type PaymentProposal struct {
+	IterationNumber            int64
+	ObservarvationPeriodTime   time.Duration
+	ObservationPeriodLedgerGap int64
+	AmountToInitiator          int64
+	AmountToResponder          int64
+	CloseSignatures            xdr.DecoratedSignature
+	DeclarationSignatures      xdr.DecoratedSignature
+}
+
+// TODO - validate inputs? (eg. no negative amounts)
+// initiator will only call this
+func (c *Channel) NewPaymentProposal(me Participant, other Participant, payToInitiator int64, payToResponder int64, startSequence int64, i int64, e int64, o time.Duration, observationPeriodLedgerGap int64, networkPassphrase string) (*PaymentProposal, error) {
```

Comment:
I think this can be simplified a little, the state machine should store state, so can track s, i, e, and o, as well as participant details.

I was thinking the caller would say how much they want to pay the other party and internally it'd work out what the new balances are. 

Wdyt?

Reply:
ya  I think you're right. I added them all for this first dirty cut but it looks like everything but pay amount can live on channel, I'll move em 👍